lib/openscap_parser/group.rb

```diff
@@ -0,0 +1,69 @@
+# frozen_string_literal: true
+module OpenscapParser
+  class Group < XmlNode
+    include OpenscapParser::Util
+
+    def id
+      @id ||= parsed_xml['id']
+    end
+
+    def title
+      @title ||= parsed_xml.at_css('title') &&
+        parsed_xml.at_css('title').text
+    end
+
+    def description
+      @description ||= newline_to_whitespace(
+        parsed_xml.at_css('description') &&
+          parsed_xml.at_css('description').text || ''
+      )
+    end
+
+    def rationale
+      @rationale ||= newline_to_whitespace(
+        parsed_xml.at_css('rationale') &&
+          parsed_xml.at_css('rationale').text || ''
+      )
+    end
+
+    def requires
+      @requires ||= parsed_xml.xpath('./requires') &&
+        parsed_xml.xpath('./requires/@idref').to_s&.split
+    end
+
+    def conflicts
+      @conflicts ||= parsed_xml.xpath('./conflicts') &&
+        parsed_xml.xpath('./conflicts/@idref').to_s
+    end
+
+    def selected
+      @selected ||= parsed_xml['selected']
+    end
+
+    def parent_id
+      @parent_id = parsed_xml.xpath('../@id').to_s
+    end
+
+    def parent_type
+      if parsed_xml.xpath("name(..)='Group'")
+        @parent_type = 'Group'
+      else
+        @parent_type = 'Benchmark'
+      end
+    end
```

Comment:
@skateman Here's how I updated it to deal with polymorphism.. So the parent_id is assigned no matter what and then the @parent_type will tell us if the parent is a `Benchmark` or `Group`